src/init.cpp

```diff
@@ -1389,16 +1389,16 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
     assert(!node.connman);
     node.connman = MakeUnique<CConnman>(GetRand(std::numeric_limits<uint64_t>::max()), GetRand(std::numeric_limits<uint64_t>::max()), args.GetBoolArg("-networkactive", true));
 
+    int ratio = std::min<int>(std::max<int>(args.GetArg("-checkmempool", chainparams.DefaultConsistencyChecks() ? 1 : 0), 0), 1000000);
+    uint32_t checkFrequency = 0;
+    if (ratio != 0) {
+    	checkFrequency = static_cast<uint32_t>((1.0/ratio) * 4294967295.0);		
```

Comment:
im not sure where this magic number comes from...? (copied from src/txmempool.h 'setSanityCheck'). I feel like there should be a comment explaining where it comes from

Reply:
4294967295 is one less than the maximum uint32 (2^32 = 4294967296). The value was originally the max uint32_t, but was changed to be one less to fix an overflow bug: https://github.com/bitcoin/bitcoin/commit/e9e616323bc44daf61f4358f5b2de58e90b5c602

Reply:
You might be able to replace it with (UINT32_MAX - 1) (see https://en.cppreference.com/w/cpp/header/cstdint)

Reply:
You could also pass the ration int into the constructor and do this int -> float -> uint32_t conversion inside the constructor.